plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -500,4 +617,13 @@ protected String getUser()
     {
         return TEST_USER;
     }
+
+    private record CastTestCase(String fromType, String castType)
+    {
+        private CastTestCase
```

Comment:
please check that `fromType != castType`
https://github.com/trinodb/trino/pull/21939/commits/edc25bf5d855ec6b82cac562d2a5407570446db7#diff-01f7e5fb80c1ce75cf0ea5d3f1933156eb7984967b99b92090bf931e93569696R1274-R1280